cmd/kube-controller-manager/app/core.go

```diff
@@ -143,14 +152,33 @@ func startNodeIpamController(ctx ControllerContext) (http.Handler, bool, error)
 		}
 	}
 
+	var nodeCIDRMaskSizeIPv4, nodeCIDRMaskSizeIPv6 int
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.IPv6DualStack) {
+		setNodeCIDRMaskSizes(ctx.ComponentConfig.NodeIPAMController, &nodeCIDRMaskSizeIPv4, &nodeCIDRMaskSizeIPv6)
+	} else {
+		setNodeCIDRMaskSizesDualStack(ctx.ComponentConfig.NodeIPAMController, &nodeCIDRMaskSizeIPv4, &nodeCIDRMaskSizeIPv6)
+	}
+
+	// IPv4 node cidr mask size not defined, set to defaultNodeMaskCIDRIPv4
+	if nodeCIDRMaskSizeIPv4 == 0 {
+		nodeCIDRMaskSizeIPv4 = defaultNodeMaskCIDRIPv4
+	}
+	// IPv6 node cidr mask size not defined, set to defaultNodeMaskCIDRIPv6
+	if nodeCIDRMaskSizeIPv6 == 0 {
+		nodeCIDRMaskSizeIPv6 = defaultNodeMaskCIDRIPv6
+	}
+
+	// get list of node cidr mask sizes
+	nodeCIDRMaskSizes := getNodeCIDRMaskSizes(clusterCIDRs, nodeCIDRMaskSizeIPv4, nodeCIDRMaskSizeIPv6)
+
 	nodeIpamController, err := nodeipamcontroller.NewNodeIpamController(
 		ctx.InformerFactory.Core().V1().Nodes(),
 		ctx.Cloud,
 		ctx.ClientBuilder.ClientOrDie("node-controller"),
 		clusterCIDRs,
 		serviceCIDR,
 		secondaryServiceCIDR,
-		int(ctx.ComponentConfig.NodeIPAMController.NodeCIDRMaskSize),
```

Comment:
:thinking:  seems we are omitting this if dualstack feature  gate is not enabled

Reply:
With this PR we are changing the default ipv6 mask size to 98. We are not omitting it.

Reply:
thanks for the clarification, you can resolve it